src/pgn.js

```diff
@@ -337,9 +355,334 @@ exports.pgnRead = function(pgnString, gameIndex) {
 				++gameCounter;
 			}
 			else {
-				throw new exception.InvalidPGN(pgnString, pgnString.length, i18n.INVALID_GAME_INDEX, gameIndex, gameCounter);
+				throw new exception.InvalidPGN(pgnString, pgnString.length, stream._lineCount, i18n.INVALID_GAME_INDEX, gameIndex, gameCounter);
 			}
 		}
 		return doParseGame(stream);
 	}
 };
+
+/**
+ * Write the date in PGN format
+ *
+ * @param {Date|{year:number, month:number}|{year:number}|undefined}} date
+ *
+ * @returns {string}
+ */
+function writeDate(date) {
+	if (date === undefined) {
+		return '[Date \'????.??.??\']' + Separator;
```

Comment:
The PGN norm requires double quotes to delimit game header values, not single quotes. See PGN format spec:
https://ia802908.us.archive.org/26/items/pgn-standard-1994-03-12/PGN_standard_1994-03-12.txt

Reply:
Oops!

Reply:
I think I screwed up when fixing for linter in one of my previous commits for linter. This is my first time writing Javascript, so pardon the mishaps!